docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,435 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Downward API for resource limits and requests
+
+## Background
+
+Currently downward API (via environment variables and volume plugin) only support exposing a Pod's name, namespace, annotations, labels and its IP ([see details](http://kubernetes.io/docs/user-guide/downward-api/)). This document explains the need and design to extend them to expose resources (e.g. cpu, memory) limits and requests.
+
+## Motivation
+
+Software applications generally configure limitations on their resource usage for their efficient operations. Exposing the requested and limited amounts of available resources inside containers will allow these applications to be configured more easily. Although docker already exposes some of this information inside containers, the downward API helps exposing this information in a runtime-agnostic manner in Kubernetes.
+
+## Design
+
+This is mostly driven by the discussion in [this issue](https://github.com/kubernetes/kubernetes/issues/9473). There are three approaches discussed in this document to obtain resources limits and requests to be exposed as environment variables and volumes inside containers:
+
+1. The first approach requires users to specify full json path selectors in which selectors are relative to the pod spec. The benefit of this approach is to specify pod level resources, and since containers are also part of a pod spec, it can be used to specify container level resources too.
+
+2. The second approach requires specifying partial json path selectors in which selectors are relative to the container spec. This approach helps in retrieving a container specific resource limits and requests, and at the same time, it is simpler to specify than full json path selectors.
+
+3. In this approach, users specify particular strings to retrieve resources limits and requests and do not specify any json path selectors. This approach is similar to the existing downward API implementation approach. The advantages of this approach is that it is simpler to specify that the firs two, and does not require any type of conversion between internal and versioned objects or json selectors as discussed below. However, unlike the full and partial path selectors approaches can be extended to other resources with no or little changes, this approach might require a relatively more work to extend them to other resources in future if needed. 
```

Comment:
The first couple sentences sound a little off.  How about:  `The third approach avoids selectors entirely and introduces a new object that has explicit fields for container name, limit type, and request type.`